src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -347,6 +347,46 @@ def get_login_credentials(self, resource=None, client_id=None, subscription_id=N
                 str(account[_TENANT_ID]))
 
     def get_raw_token(self, resource=None, scopes=None, subscription=None, tenant=None):
+        # Check if AZ_USE_AZD_AUTH environment variable is set
+        if os.environ.get('AZ_USE_AZD_AUTH'):
```

Comment:
Only changing `get_raw_token` is far from enough.

1. `get_raw_token` is only designed for getting the raw token. It cannot be used with Python SDK. To invoke Python SDK, it is required to call `get_login_credentials` to get the credential.
2. The account (returned by `az account show`) containing the current subscription and tenant must also be set. This makes `get_subscription` work, which is required by `get_login_credentials`: https://github.com/Azure/azure-cli/blob/9091fee18ad8f3e82b18115fb1a3d762b871d5dc/src/azure-cli-core/azure/cli/core/_profile.py#L308